### PR TITLE
feat: Add OneToOneFPMM for stablecoin swaps

### PR DESCRIPTION
**Pull request by bowd**
### Description

This PR introduces the `OneToOneFPMM` contract, a specialized Fixed Price Market Maker designed for stablecoin-to-stablecoin swaps (e.g., USD.m ↔ USDC/USDT). Unlike the standard FPMM which uses oracle rates, the OneToOneFPMM always swaps at a 1:1 rate regardless of oracle pricing, making it ideal for swaps between stablecoins pegged to the same asset.

**Current behavior:**
- FPMMs use oracle rates to determine swap prices
- Oracle adapter checks FX market hours, rate freshness, and trading mode before providing rates

**Updated behavior with this PR:**
- OneToOneFPMM always swaps at exactly 1:1 ratio
- Validates rate feed using new `ensureRateValid()` method which checks:
  - Trading mode (breaker box suspension)
  - Rate freshness (NO_RECENT_RATE check)
- Does not load actual oracle rate data (which would be ignored anyway)
- Works 24/7 regardless of FX market hours (no FX market check)
- Requires fresh oracle data to ensure oracle is functioning

### Other changes

1. **Renamed `revertRateFeed` to `invertRateFeed`** in FPMM interfaces and implementation
   - More accurately describes the parameter's purpose
   - Added `invertRateFeed` parameter to `FPMMFactory.deployFPMM()` methods

2. **Added `IOracleAdapter.ensureRateValid()`** method
   - Efficiently validates rate feed without loading oracle rate data
   - Checks trading mode (breaker box) and rate freshness
   - Reverts with appropriate error messages if validation fails

### Tested

- **Unit tests for OneToOneFPMM**: 19 tests covering swap operations, amount calculations, and edge cases
  - Tests verify 1:1 swaps work regardless of oracle rates
  - Tests verify trading suspension is respected
  - Tests verify swaps work when FX markets are closed
  - Tests verify swaps revert with stale oracle data (NO_RECENT_RATE)
  
- **OracleAdapter tests**: Added 3 tests for `ensureRateValid()` method
  - Trading suspended case
  - No recent rate case  
  - Valid rate case

- **Updated FPMM Factory tests**: Updated 11 test callsites to include new `invertRateFeed` parameter

All 45+ tests pass successfully.

### Related issues

- Part of the ongoing stablecoin infrastructure improvements

### Backwards compatibility

**Breaking changes:**
- `FPMMFactory.deployFPMM()` now requires an additional `bool invertRateFeed` parameter
- Any external contracts or scripts calling `deployFPMM()` will need to be updated

**Non-breaking changes:**
- `IOracleAdapter.ensureRateValid()` is a new method that doesn't affect existing functionality
- `OneToOneFPMM` is a new contract that doesn't modify existing FPMM behavior
- Renaming `revertRateFeed` to `invertRateFeed` is internal to the FPMM system

### Documentation

No external documentation changes required at this time. The contracts contain comprehensive natspec comments explaining the new functionality.

### CODE REVIEW

File: test/unit/swap/OneToOneFPMM/OneToOneFPMMGetAmountOut.t.sol [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+    // Even though oracle says 2:1, OneToOneFPMM always returns 1:1 (minus fee)
+    uint256 expectedAmountOut = 99.7e18; // 100e18 - 0.3% fee
+    assertEq(amountOut, expectedAmountOut);
+
+    amountOut = fpmm.getAmountOut(amountIn, token1);
+    assertEq(amountOut, expectedAmountOut);
+  }
+
+  function test_getAmountOut_whenOracleRateDiffers_shouldIgnoreOracleAndUse1to1()
```
Comment (nvtaveras) [inline test/unit/swap/OneToOneFPMM/OneToOneFPMMGetAmountOut.t.sol:41]: This test is identical to the above (`test_getAmountOut_whenRateIsAlways1to1_shouldReturnCorrectAmount`) except for the slightly different rate. Let's get rid of one of them?

File: test/unit/swap/OneToOneFPMM/OneToOneFPMMSwap.t.sol [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  function test_swap_whenRateIsExpired_shouldRevert()
+    public
+    initializeFPMM_withDecimalTokens(18, 18)
+    mintInitialLiquidity(18, 18)
+    withOracleRate(1e18, 1e18)
+    withFXMarketOpen(true)
+    withRecentRate(false)
+  {
+    // OneToOneFPMM checks rate freshness via ensureRateValid
+    deal(token0, address(fpmm), 100e18);
```
Comment (nvtaveras) [inline test/unit/swap/OneToOneFPMM/OneToOneFPMMSwap.t.sol:177]: ditto

**Review by nvtaveras — APPROVED** [review]
LGTM, just some minor comments
